src/modules/cpu/kernel/slice.hpp

```diff
@@ -26,184 +26,210 @@ SOFTWARE.
+            if((srcGenericDescPtr->layout == RpptLayout::NCDHW) && (dstGenericDescPtr->layout == RpptLayout::NCDHW))
             {
-                Rpp32f *srcPtrDepth, *dstPtrDepth;
-                srcPtrDepth = srcPtrChannel;
-                dstPtrDepth = dstPtrChannel;
+                srcPtrChannel = srcPtrTemp + (anchor[1] * srcGenericDescPtr->strides[2]) + (anchor[2] * srcGenericDescPtr->strides[3]) + (anchor[3] * layoutParams.bufferMultiplier);
+                Rpp32u maxDepth = std::min(shape[1], length[1] - anchor[1]);
+                Rpp32u maxHeight = std::min(shape[2], length[2] - anchor[2]);
+                Rpp32u maxWidth = std::min(shape[3], length[3] - anchor[3]);
+                Rpp32u bufferLength = maxWidth * layoutParams.bufferMultiplier;
+                Rpp32u copyLengthInBytes = bufferLength * sizeof(T);
+
+                // if padding is required, fill the buffer with fill value specified
+                bool needPadding = (((anchor[1] + shape[1]) > length[1]) ||
+                                    ((anchor[2] + shape[2]) > length[2]) ||
+                                    ((anchor[3] + shape[3]) > length[3]));
+                if(needPadding && enablePadding)
+                    std::fill(dstPtrChannel, dstPtrChannel + dstGenericDescPtr->strides[0] - 1, *fillValue);
+
+                for(int c = 0; c < layoutParams.channelParam; c++)
+                {
+                    T *srcPtrDepth, *dstPtrDepth;
+                    srcPtrDepth = srcPtrChannel;
+                    dstPtrDepth = dstPtrChannel;
+                    for(int i = 0; i < maxDepth; i++)
+                    {
+                        T *srcPtrRow, *dstPtrRow;
+                        srcPtrRow = srcPtrDepth;
+                        dstPtrRow = dstPtrDepth;
+                        for(int j = 0; j < maxHeight; j++)
+                        {
+                            memcpy(dstPtrRow, srcPtrRow, copyLengthInBytes);
+                            srcPtrRow += srcGenericDescPtr->strides[3];
+                            dstPtrRow += dstGenericDescPtr->strides[3];
+                        }
+                        srcPtrDepth += srcGenericDescPtr->strides[2];
+                        dstPtrDepth += dstGenericDescPtr->strides[2];
+                    }
+                    srcPtrChannel += srcGenericDescPtr->strides[1];
+                    dstPtrChannel += srcGenericDescPtr->strides[1];
+                }
+            }
 
-                for(int i = 0; i < roi.xyzwhdROI.roiDepth; i++)
+            // slice without fused output-layout toggle (NDHWC -> NDHWC)
+            else if((srcGenericDescPtr->layout == RpptLayout::NDHWC) && (dstGenericDescPtr->layout == RpptLayout::NDHWC))
+            {
+                srcPtrChannel = srcPtrTemp + (anchor[0] * srcGenericDescPtr->strides[1]) + (anchor[1] * srcGenericDescPtr->strides[2]) + (anchor[2] * layoutParams.bufferMultiplier);
```

Comment:
L109-L121 are the same for NCDHW and NDHWC if you pass a hardcoded int3 or a tuple (0, 1, 2) for the second (1, 2, 3) for the first. Same comment for numDims = 3. 
Code for 2 and 1 seems short enough.

Same comment for HIP.